server/odc-service/src/main/java/com/oceanbase/odc/service/git/repo/RepositoryManager.java

```diff
@@ -0,0 +1,212 @@
+import com.google.common.base.MoreObjects;
+import com.oceanbase.odc.common.crypto.TextEncryptor;
+import com.oceanbase.odc.common.lang.Pair;
+import com.oceanbase.odc.common.util.StringUtils;
+import com.oceanbase.odc.common.util.SystemUtils;
+import com.oceanbase.odc.core.shared.Verify;
+import com.oceanbase.odc.core.shared.constant.ResourceType;
+import com.oceanbase.odc.core.shared.exception.NotFoundException;
+import com.oceanbase.odc.metadb.git.GitRepoRepository;
+import com.oceanbase.odc.metadb.git.GitRepoStageRepository;
+import com.oceanbase.odc.metadb.git.GitRepositoryEntity;
+import com.oceanbase.odc.metadb.git.GitRepositoryStageEntity;
+import com.oceanbase.odc.service.encryption.EncryptionFacade;
+import com.oceanbase.odc.service.git.GitClientOperator;
+import com.oceanbase.odc.service.git.GitIntegrationService;
+import com.oceanbase.odc.service.git.model.GitIntegrationProperties;
+import com.oceanbase.odc.service.git.model.RepoState;
+import com.oceanbase.odc.service.iam.auth.AuthenticationFacade;
+import com.oceanbase.odc.service.objectstorage.cloud.CloudObjectStorageService;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author: liuyizhuo.lyz
+ * @date: 2024/8/7
+ */
+@Slf4j
+@Component
+public class RepositoryManager {
+    private static final String LOCK_FORMAT = "repo-lock-%s-%s";
+    private static final String GIT_REPO_DIR =
+            Paths.get(MoreObjects.firstNonNull(SystemUtils.getEnvOrProperty("file.storage.dir"), "./data"), "repos")
+                    .toString();
+
+    @Autowired
+    private GitRepoRepository gitRepoRepository;
+    @Autowired
+    private GitRepoStageRepository stageRepository;
+    @Autowired
+    private EncryptionFacade encryptionFacade;
+    @Autowired
+    private AuthenticationFacade authenticationFacade;
+    @Autowired
+    private CloudObjectStorageService objectStorageService;
+    @Autowired
+    private JdbcLockRegistry jdbcLockRegistry;
+    @Autowired
+    private GitIntegrationService integrationService;
+
+    /**
+     * Use {@code repoId}、{@code userId} to identify repository
+     */
+    private final LoadingCache<Pair<Long, Long>, GitClientOperator> operatorCache;
+
+    public RepositoryManager(@Autowired GitIntegrationProperties gitIntegrationProperties) {
+        operatorCache = Caffeine.newBuilder()
+                .maximumSize(gitIntegrationProperties.getGitRepositoryMaxCachedSize())
+                .expireAfterAccess(Duration.ofMinutes(gitIntegrationProperties.getGitRepositoryPreserveMinutes()))
+                .removalListener((pair, operator, cause) -> {
+                    if (operator != null) {
+                        deleteRepo((GitClientOperator) operator);
+                        ((GitClientOperator) operator).close();
+                    }
+                })
+                .build(this::createGitOperator);
+    }
+
+    @PostConstruct
+    public void init() {
+        try {
+            FileUtils.cleanDirectory(new File(GIT_REPO_DIR));
```

Comment:
Why is it necessary to clean this directory every time ODC restarts? Is this directory intended to be a temporary directory?

Reply:
yes. It's a temporary dir. User's repository would be removed regularly. 
And the remove operations were managed in memory. So if we do not clean the directory when ODC restarts, there maybe some

Reply:
But what if the ODC never restarts or restarts very infrequently? Shouldn't we have a scheduled cleanup mechanism instead of relying on the ODC restart to clean the temporary directory?

Reply:
I dont think clean up periodically is a good choice. A one-time cleaning is simple, stable, and meets the needs of our business. 
Even if ODC restarts frequently, the cost of cleaning temporary folders is not very high. What are your specific concerns about this place?

Reply:
I mean **infrequently**, not **frequently**. If your purpose is to clean up the directory to ensure that some unnecessary folders are eventually deleted, then your 'one-time cleaning' approach cannot guarantee this. Conversely, if you don't care about the temp directory, why do we need this cleanup operation at all?

Reply:
... Sorry I misread. Then this question is simple. Each GitOperator in the cache will expire after 24 hours. At the same time as this expires, the workspace directory will be cleaned.
So you can think of it as regular cleaning